R/symptome_compartment_class.R

```diff
@@ -0,0 +1,226 @@
+                           paste(names(object@initial_population)[is_na_pop], collapse = ", ")
+              )
+              errors <- c(errors, msg)
+            }else{
+              # check whether the sum of initial population is normalized to 1
+              sum_init_pop <- sum(unlist(object@initial_population))
+              if (sum_init_pop != 1) {
+                msg <- "Sum of initial population is not 1, please normalize"
+                errors <- c(errors, msg)
+              }
+            }
+            # check whether the lengths of nu and pdeath correspond to the number of infected groups
+            n_nu <- length(object@parameters$nu)
+            if (n_nu != 2) {
+              msg <- paste0(
+                "Length of parameter nu,", n_nu, ", is not equal to the setting ", 2)
+              errors <- c(errors, msg)
+            }
+            n_pdeath <- length(object@parameters$pdeath)
+            if (n_pdeath != 2) {
+              msg <- paste0(
+                "Length of parameter pdeath,", n_pdeath, ", is not equal to the setting ", 2)
+              errors <- c(errors, msg)
+            }
+            if (length(errors) == 0) TRUE else errors
+          })
+
+
+#' @describeIn model_symptome to solve the ode system.
+#'
+#' @param object An object of class model_symptome
+#' @param times A list of time points of the simulation period
+#' @param method A string indicating which ode integrator to use. Default is set to 'lsoda'
+#' @return An object of class model_symptome with a dataframe of simulation output
+#'
+setGeneric(name = "ode_simulate",
+           def = function(object, times, method = "lsoda") standardGeneric("ode_simulate"))
+
+setMethod("ode_simulate", signature(object = "model_symptome"),
+  function(object, times, method) {
+    # initial population groups
+    pop_groups <- c(S = object@initial_population$S,
+                    E = object@initial_population$E,
+                    I = object@initial_population$I,
+                    HA = object@initial_population$HA,
+                    R = object@initial_population$R,
+                    CM = object@initial_population$CM)
+    # parameters
+    params <- c(lam = object@parameters$lam,
+                gamma = object@parameters$gamma,
+                omega = object@parameters$omega,
+                ihr = object@parameters$ihr,
+                nu = object@parameters$nu,
+                pdeath = object@parameters$pdeath)
+
+    # ODE system RHS
+    ode_symptome_rhs <- function(t, pop_groups, parameters) {
+      with(
+        as.list(c(pop_groups, params)), {
+          dS <- -lam * S * (I + HA) + omega * R
+          dE <- lam * S * (I + HA) - gamma * E
+          dI <- (1.0 - ihr) * gamma * E - nu.i * I  - pdeath.i * I
+          dHA <- ihr * gamma * E - nu.hsa * HA  - pdeath.hsa * HA
+          dR <- -omega * R + nu.i * I + nu.hsa * HA
+          dCM <-  pdeath.i * I + pdeath.hsa * HA
```

Comment:
I think "I" is mild infection? I would have thought it might be more reflective of the Como model to have three compartments: `I_asymptomatic`, `I_mild`, `I_hospitalised` (I'd call them those names too so that people know what they are). Then you could just define `I_total = I_asymptomatic + I_mild + I_hospitalised` which you'd use in the `S` and `E` equations? Similarly, I'd call `CM` -> `D_cumulative` to make it more understandable.

Reply:
In terms of naming of compartments, I totally agree that what you suggest are more readable and user-friendly.  Right now I am taking the names from the como code for both the compartments and the transmission parameters. So just to confirm, we don't necessarily want to keep the names of the compartments same with the current como code, do we?